diff -up net-snmp-5.9.4/agent/mibgroup/if-mib/data_access/interface_linux.c.3~ net-snmp-5.9.4/agent/mibgroup/if-mib/data_access/interface_linux.c
--- net-snmp-5.9.4/agent/mibgroup/if-mib/data_access/interface_linux.c.3~	2023-08-16 13:53:00.832615468 +0200
+++ net-snmp-5.9.4/agent/mibgroup/if-mib/data_access/interface_linux.c	2023-08-16 13:56:23.395556577 +0200
@@ -35,7 +35,7 @@ static struct pci_access *pci_access;
 static int do_longjmp =0;
 static jmp_buf err_buf;
 PCI_NONRET static void
-netsnmp_pci_error(char *msg, ...)
+netsnmp_pci_error(char *msg, ...) __attribute__((noreturn))
 {
     va_list args;
     char *buf;
