diff --git a/agent/snmp_perl.c b/agent/snmp_perl.c
index 48d9695..59fb534 100644
--- a/agent/snmp_perl.c
+++ b/agent/snmp_perl.c
@@ -5,6 +5,10 @@
 #include <EXTERN.h>
 #include "perl.h"
 
+#ifdef U64TYPE
+#define U64 U64pairU32
+#endif
+
 #include <net-snmp/net-snmp-config.h>
 #include <net-snmp/net-snmp-includes.h>
 #include <net-snmp/agent/net-snmp-agent-includes.h>
diff --git a/perl/ASN/Makefile.PL b/perl/ASN/Makefile.PL
index 4576781..4617d46 100644
--- a/perl/ASN/Makefile.PL
+++ b/perl/ASN/Makefile.PL
@@ -1,3 +1,4 @@
+package ASN;
 use ExtUtils::MakeMaker;
 require 5;
 use Config;
diff --git a/perl/AnyData_SNMP/Makefile.PL b/perl/AnyData_SNMP/Makefile.PL
index 8106650..e92e012 100644
--- a/perl/AnyData_SNMP/Makefile.PL
+++ b/perl/AnyData_SNMP/Makefile.PL
@@ -1,3 +1,4 @@
+package AnyData_SNMP;
 use ExtUtils::MakeMaker;
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
diff --git a/perl/OID/Makefile.PL b/perl/OID/Makefile.PL
index 6bb1616..7905445 100644
--- a/perl/OID/Makefile.PL
+++ b/perl/OID/Makefile.PL
@@ -1,3 +1,4 @@
+package OID;
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
 
diff --git a/perl/OID/OID.xs b/perl/OID/OID.xs
index 10485f9..3bfb4ca 100644
--- a/perl/OID/OID.xs
+++ b/perl/OID/OID.xs
@@ -7,6 +7,10 @@
 #include "perl.h"
 #include "XSUB.h"
 
+#ifdef U64TYPE
+#define U64 U64pairU32
+#endif
+
 #include <net-snmp/net-snmp-config.h>
 #include <net-snmp/net-snmp-includes.h>
 
diff --git a/perl/SNMP/Makefile.PL b/perl/SNMP/Makefile.PL
index e617cb7..e3d11d7 100644
--- a/perl/SNMP/Makefile.PL
+++ b/perl/SNMP/Makefile.PL
@@ -1,3 +1,4 @@
+package SNMP;
 use ExtUtils::MakeMaker;
 require 5;
 use Config;
diff --git a/perl/SNMP/SNMP.xs b/perl/SNMP/SNMP.xs
index 2a734e8..230ffac 100644
--- a/perl/SNMP/SNMP.xs
+++ b/perl/SNMP/SNMP.xs
@@ -16,6 +16,10 @@
 #include "perl.h"
 #include "XSUB.h"
 
+#ifdef U64TYPE
+#define U64 U64pairU32
+#endif
+
 #include <net-snmp/net-snmp-config.h>
 #include <net-snmp/net-snmp-includes.h>
 #include <sys/types.h>
diff --git a/perl/TrapReceiver/Makefile.PL b/perl/TrapReceiver/Makefile.PL
index 874ee21..680613b 100644
--- a/perl/TrapReceiver/Makefile.PL
+++ b/perl/TrapReceiver/Makefile.PL
@@ -1,3 +1,4 @@
+package TrapReceiver;
 use ExtUtils::MakeMaker;
 require 5;
 use Config;
diff --git a/perl/TrapReceiver/TrapReceiver.xs b/perl/TrapReceiver/TrapReceiver.xs
index a4fc136..3364509 100644
--- a/perl/TrapReceiver/TrapReceiver.xs
+++ b/perl/TrapReceiver/TrapReceiver.xs
@@ -9,6 +9,10 @@
 
 #include "ppport.h"
 
+#ifdef U64TYPE
+#define U64 U64pairU32
+#endif
+
 #include <net-snmp/net-snmp-config.h>
 #include <net-snmp/net-snmp-includes.h>
 #include <net-snmp/agent/net-snmp-agent-includes.h>
diff --git a/perl/agent/Makefile.PL b/perl/agent/Makefile.PL
index 4f7bee3..b7e42e9 100644
--- a/perl/agent/Makefile.PL
+++ b/perl/agent/Makefile.PL
@@ -1,3 +1,4 @@
+package agent;
 use ExtUtils::MakeMaker;
 require 5;
 use Config;
diff --git a/perl/agent/Support/Makefile.PL b/perl/agent/Support/Makefile.PL
index 48815b6..4b2a1ff 100644
--- a/perl/agent/Support/Makefile.PL
+++ b/perl/agent/Support/Makefile.PL
@@ -1,3 +1,4 @@
+package agent_Support;
 use ExtUtils::MakeMaker;
 require 5;
 use Config;
diff --git a/perl/agent/agent.xs b/perl/agent/agent.xs
index c7b5d9a..bb36a7b 100644
--- a/perl/agent/agent.xs
+++ b/perl/agent/agent.xs
@@ -9,6 +9,10 @@
 #include <netdb.h>
 #include <sys/socket.h>
 
+#ifdef U64TYPE
+#define U64 U64pairU32
+#endif
+
 #include <net-snmp/net-snmp-config.h>
 #include <net-snmp/net-snmp-includes.h>
 #include <net-snmp/agent/net-snmp-agent-includes.h>
diff --git a/perl/agent/default_store/Makefile.PL b/perl/agent/default_store/Makefile.PL
index ed6483a..569abd4 100644
--- a/perl/agent/default_store/Makefile.PL
+++ b/perl/agent/default_store/Makefile.PL
@@ -1,3 +1,4 @@
+package agent_default_store;
 use ExtUtils::MakeMaker;
 require 5;
 use Config;
diff --git a/perl/default_store/Makefile.PL b/perl/default_store/Makefile.PL
index 7c671b8..952e294 100644
--- a/perl/default_store/Makefile.PL
+++ b/perl/default_store/Makefile.PL
@@ -1,3 +1,4 @@
+package default_store;
 use ExtUtils::MakeMaker;
 require 5;
 use Config;
diff --git a/perl/manager/Makefile.PL b/perl/manager/Makefile.PL
index 633243d..2f889a1 100644
--- a/perl/manager/Makefile.PL
+++ b/perl/manager/Makefile.PL
@@ -1,3 +1,4 @@
+package manager;
 use ExtUtils::MakeMaker;
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
