diff -up net-snmp-5.9.1/testing/fulltests/default/T070com2sec_simple.2~ net-snmp-5.9.1/testing/fulltests/default/T070com2sec_simple
--- net-snmp-5.9.1/testing/fulltests/default/T070com2sec_simple.2~	2022-07-09 22:51:22.419182035 +0200
+++ net-snmp-5.9.1/testing/fulltests/default/T070com2sec_simple	2022-07-09 22:52:07.110987867 +0200
@@ -80,6 +80,10 @@ CONFIGAGENT 'com2sec t407a no.such.addre
 # 407b Test maximally long host name/mask
 CONFIGAGENT 'com2sec t407b a23456789012345678901234567890123456789012345678901234567890123.a23456789012345678901234567890123456789012345678901234567890123.a23456789012345678901234567890123456789012345678901234567890123.a234567890123456789012345678901234567890123456789012345678901./255.255.255.255 c407b'
 
+FINISHED
+
+# don't test the later, it depends on DNS, which is not available in ABF
+
 # 408 Lookup tests, require network access
 # 408a Test lookup returning a single host
 CONFIGAGENT 'com2sec t408a onea.net-snmp.org c408a'
diff -up net-snmp-5.9.1/testing/fulltests/default/T071com2sec6_simple.2~ net-snmp-5.9.1/testing/fulltests/default/T071com2sec6_simple
--- net-snmp-5.9.1/testing/fulltests/default/T071com2sec6_simple.2~	2022-07-09 22:51:22.419182035 +0200
+++ net-snmp-5.9.1/testing/fulltests/default/T071com2sec6_simple	2022-07-09 22:52:48.521881298 +0200
@@ -74,6 +74,10 @@ CONFIGAGENT 'com2sec6 t607a no.such.addr
 # 607b Test maximally long host name/mask
 CONFIGAGENT 'com2sec6 t607b a23456789012345678901234567890123456789012345678901234567890123.a23456789012345678901234567890123456789012345678901234567890123.a23456789012345678901234567890123456789012345678901234567890123.a234567890123456789012345678901234567890123456789012345678901./ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff c607b'
 
+FINISHED
+
+# don't test the later, it depends on DNS, which is not available in ABF
+
 # 608 Lookup tests, require network access
 # 608a Test lookup returning a single host
 CONFIGAGENT 'com2sec6 t608a oneAAAA.net-snmp.org c608a'
